File: cpp/leap/leap.cpp
#include <iostream>

namespace leap {

bool is_leap_year(int int_year)
    {
        // std::cout<<"checking if "<< int_year <<" is a leap year\n";

        // not that readable, but it's in a single line
        return (int_year % 4 == 0) && (!(int_year % 100 == 0) || (int_year % 400 == 0));
    };
};

File: python/leap/leap.py
def is_leap_year(year):
    # if the year is divisible by 4 AND if it's either not divisible by 100 or divisible by 400,
    # then it's a leap year
    return (year % 4 == 0) and (not (year % 100 == 0) or (year % 400 == 0))

